objects/src/accounts/delta/vault.rs

```diff
@@ -148,6 +147,32 @@ impl AccountVaultDelta {
     }
 }
 
+impl From<&AssetVault> for AccountVaultDelta {
+    fn from(vault: &AssetVault) -> Self {
+        let mut fungible = BTreeMap::new();
+        let mut non_fungible = BTreeMap::new();
+
+        for asset in vault.assets() {
+            match asset {
+                Asset::Fungible(asset) => {
+                    fungible.insert(
+                        asset.faucet_id(),
+                        asset.amount().try_into().expect("Amount must fit into `i64`"),
```

Comment:
```suggestion
                        asset.amount().try_into().expect("asset amount should be at most i64::MAX by construction"),
```

Nit: Lowercase message and `expect` messages ideally use "should".